src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/HealthCertificate+PDF/HealthCertificate+PDF.swift

```diff
@@ -0,0 +1,331 @@
+			fatalError("Could not load pdf template.")
+		}
+		return templateData
+	}
+
+	private func texts(with valueSets: SAP_Internal_Dgc_ValueSets) -> [PDFText] {
+		switch entry {
+		case .vaccination(let entry):
+			return vaccinationTexts(for: entry, with: valueSets)
+		case .test(let entry):
+			return testTexts(for: entry, with: valueSets)
+		case .recovery(let entry):
+			return recoveryTexts(for: entry, with: valueSets)
+		}
+	}
+
+	private func vaccinationTexts(for entry: VaccinationEntry, with valueSets: SAP_Internal_Dgc_ValueSets) -> [PDFText] {
+		[
+			PDFText(
+				text: entry.certificateIssuer,
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 19, y: 30, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.countryOfVaccination,
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 19, y: 61, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.dateOfVaccination,
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 19, y: 93, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.formattedValue(for: \VaccinationEntry.marketingAuthorizationHolder, valueSets: valueSets) ?? "",
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 158, y: 173, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.formattedValue(for: \VaccinationEntry.vaccineMedicinalProduct, valueSets: valueSets) ?? "",
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 158, y: 230, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.formattedValue(for: \VaccinationEntry.vaccineOrProphylaxis, valueSets: valueSets) ?? "",
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 19, y: 303, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.formattedValue(for: \VaccinationEntry.diseaseOrAgentTargeted, valueSets: valueSets) ?? "",
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 19, y: 331, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: "\(entry.doseNumber)  \(entry.totalSeriesOfDoses)",
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 19, y: 141, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: name.reversedFullNameWithoutFallback,
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 28, y: 694, width: 266, height: 23),
+				upsideDown: false
+			),
+			PDFText(
+				text: dateOfBirth,
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 28, y: 740, width: 266, height: 23),
+				upsideDown: false
+			),
+			PDFText(
+				text: entry.uniqueCertificateIdentifier,
```

Comment:
In the tech spec I see something about a prefix that is supposed to be cut off, is that missing? https://github.com/corona-warn-app/cwa-app-tech-spec/pull/92/files#diff-25c8b7188514d7cdfb430f7f8f963fed63bca279fd74753bc6cefe29dd599b55R1328

Reply:
good catch.  Done